src/components/Form/Integer.js

```diff
@@ -48,6 +48,17 @@ const fixInt = (value, oldValue, keyCode) => {
   return '0';
 };
 
+const isInputValid = (allowedKeys, keyCode, targetValue, validRegex) => {
+  if (allowedKeys.includes(keyCode)) {
+    return !(
+      targetValue === '' &&
+      (keyCode === KEY_CODES[0] || keyCode === KEY_CODES.NUMPAD[0]) &&
```

Comment:
This is just a specific optimization (only first value of positive int), and does not concern other use cases.
What is the desired use case of this? This functionality is also tackled by timeouted validRegex (is it missing something?).